main.py

```diff
@@ -1152,3 +1152,45 @@ def handle_link_maintenance_end(self, event):
             link.endpoint_a.enable()
             link.endpoint_b.enable()
             self.notify_link_status_change(link, reason='maintenance')
+
+    @listen_to('topology.interruption.start')
+    def on_interruption_start(self, event: KytosEvent):
+        """Deals with the start of service interruption."""
+        with self._links_lock:
+            self.handle_interruption_start(event)
+
+    def handle_interruption_start(self, event: KytosEvent):
+        """Deals with the start of service interruption."""
+        interrupt_type = event.content['type']
+        # switches = event.content.get('switches', [])
+        # interfaces = event.content.get('interfaces', [])
+        links = event.content.get('links', [])
+        # for switch_id in switches:
+        #     pass
+        # for interface_id in interfaces:
+        #     pass
+        for link_id in links:
+            link = self.links[link_id]
+            self.notify_link_status_change(link, interrupt_type)
```

Comment:
Same comment as below (check out that one first where I elaborated on the reason why). Could you add a safe guard `link.status == EntityStatus.DOWN and len(link.status_reason) == 1`? Just so it only sends a `kytos/topology.link_down` for the first interruption, since `kytos/topology.link_down` is a higher level event.

Reply:
The problem is that we might potentially miss a link_down event, if we put guards around it like this. We could have a situation where multiple interruptions occur on a single device at the same time, resulting in multiple values in `status_reason` before we check its current state.

What we could do is have a set of links which we have sent out state messages for, and use that to ensure consistency.

Reply:
Indeed, @Ktmi. Right. Yes, Keeping track of the last status for each link and only send the event if it's different. 

`self.notify_link_status_change` is being called in multiple places currently, maybe we could evolve it into `self.try_to_notify_link_status_change` where it'd have this statefullness with a dict that you can introduce, we'd be increasing it's responsibility at the expense of keeping the logic in a single place, seems great to me. We'll need lock protection there too to be safe.

I also think that we can get away without having to store this extra last state in the database and only keep it at runtime since during start up when a switch and its interfaces are loaded [interfaces will always be deactivated](https://github.com/kytos-ng/topology/blob/master/main.py#L181) since it always driven by the protocol since it's not safe to assume that after a restart it'd be active again. So, this dict could assume that the initial status is `DOWN` during a startup. With that said, this would need a bit of exploratory test since link notification is used in many places, maybe we can deliver this PR with potential  redudant `kytos/topology.link_up|link_down` and only address this part later on `2023.1.1` just so we have proper time to explore this part in a new PR? Wdyt? 

With this approach there's also opportunity to remove `last_status_is_active` but that one let's leave just for `2023.2` to avoid increasing the scope: https://github.com/kytos-ng/topology/issues/122 https://github.com/kytos-ng/topology/issues/122